site/cloudbuild.yaml

```diff
@@ -29,6 +29,8 @@ steps:
   args: ['build', '-f', 'Dockerfile.build', '-t', 'make-docker', '.'] # we need docker and make to run everything.
 - name: "make-docker"
   dir: "build"
+  env:
+    - "REGISTRY=${_REGISTRY}"
```

Comment:
🤔 what does this do? I don't _think_ this is required, but I could be wrong?

Reply:
So this all comes from here:
https://github.com/googleforgames/agones/blob/176ffb21af4d621d8afd310b33e3e10b627d7bc1/build/includes/build-image.mk#L51

So pull and push build image targets are targets specifically for the CI system (local developers don't use them), so for those, I don't know if you want to set them to a default that will work on CI?

Also, if you have `_REGISTRY` you will need to specify it in the substitutions block of this cloudbuild.yaml

I'm curious if you tested running this cloud-build yaml against your dev project to see if it worked - because I would have expected it would have failed because of the missing `_REGISTRY` definition.

Reply:
Right, I understood they come from the `build-image.mk`. I'm just not sure where the [`REGISTRY`](https://github.com/googleforgames/agones/blob/176ffb21af4d621d8afd310b33e3e10b627d7bc1/build/includes/build-image.mk#L22) come from in this file. So if we do not specify the `REGISTRY` env viable in the `site/cloud-build.yaml`, will it try to pull the image from `gcr.io`?

You're right the `_REGISTRY` definition is missed. I only noticed it when I saw your last comments.

Reply:
OIC OIC! So the `REGISTRY` variable in the Makefile could be set in the parent Makefile or any includes that came before it.

So in the `main` version of the code, it come from here: https://github.com/googleforgames/agones/blob/176ffb21af4d621d8afd310b33e3e10b627d7bc1/build/Makefile#L40

>  So if we do not specify the REGISTRY env viable in the site/cloud-build.yaml, will it try to pull the image from gcr.io?

So in your code, the `REGISTRY` variable is removed from the parent `Makefile`, so I'm not sure where it would be getting it from -- I actually expect that it would fail silently, since there wouldn't be a registry attached.

https://github.com/googleforgames/agones/blob/176ffb21af4d621d8afd310b33e3e10b627d7bc1/build/includes/build-image.mk#L70-L72

So TL;DR - I would suggest either hardcode where it's pulled from somewhere in that Makefile, or create a substitution section like we have in the main cloudbuild.yaml and pull it from there.

Reply:
Thanks for the details! This makes sense now.

Reply:
Added the substitution in `site/cloudbuild.yaml`. Confirmed the cloud build step is looking at the `us-docker.pkg.dev/${PROJECT_ID}/ci`, and the corresponding `agones-build` image exists in `us-docker.pkg.dev/agones-images/ci`.

Reply:
Looks right! Okay, let's approve and hopefully nothing breaks 😀 

But looks good to me!